ingest/indexer/indexer_config.go

```diff
@@ -0,0 +1,59 @@
+package indexer
+
+import (
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+
+	"github.com/osmosis-labs/osmosis/osmoutils"
+	"github.com/osmosis-labs/osmosis/v25/ingest/indexer/domain"
+	service "github.com/osmosis-labs/osmosis/v25/ingest/indexer/service/client"
+)
+
+// Config defines the config for the indexer.
+type Config struct {
+	IsEnabled                bool   `mapstructure:"enabled"`
+	GCPProjectId             string `mapstructure:"gcp-project-id"`
+	BlockTopicId             string `mapstructure:"block-topic-id"`
+	TransactionTopicId       string `mapstructure:"transaction-topic-id"`
+	PoolTopicId              string `mapstructure:"pool-topic-id"`
+	TokenSupplyTopicId       string `mapstructure:"token-supply-topic-id"`
+	TokenSupplyOffsetTopicId string `mapstructure:"token-supply-offset-topic-id"`
+}
```

Comment:
We also need to add logic for generating the default config when we do `osmosisd init`.

It doesn't seem to me that that logic is present. I suspect that's the case for SQS as well.

Let's discuss doing it in this PR vs follow-up

Reply:
Yes I thought about that. Just not sure what default we should have - maybe making them all pointing to dev settings?

Reply:
I think we can default to disabled and empty strings/zeroes

Let's make sure we track this via issues 🙏

Reply:
[Issue created](https://linear.app/osmosis/issue/IND-51/added-defaultconfig-for-sqs-and-indexer-when-osmosisd-init)